connect/file/src/main/java/org/apache/kafka/connect/file/FileStreamSinkConnector.java

```diff
@@ -66,4 +70,14 @@ public void start(Map<String, String> props) {
     public void stop() {
         // Nothing to do since FileStreamSinkConnector has no background monitoring.
     }
+
+    @Override
+    public ConfigDef defineConfig() {
+        if (this.configDef != null) {
```

Comment:
Is every connector going to end up having this same logic? Could the default implementation of `Connector.validate()` just call `defineConfig()` instead of using the field? This does leave the risk of an inefficient implementation always creating the `ConfigDef` inside `defineConfig` instead of creating it once and reusing it, but that doesn't seem like it'd ever be a real performance concern so the simplicity might be worth it.